src/console_cmds.cpp

```diff
@@ -716,6 +716,11 @@ DEF_CONSOLE_CMD(ConClientNickChange)
 		return true;
 	}
 
+	if (!NetworkIsValidClientName(argv[2])) {
+		IConsoleError("Cannot give a client an invalid name");
```

Comment:
Can we make this error a bit nicer? At least give some snippets what "invald" is? It now sounds so harsh :D

Reply:
I'm a bit conflicted about this. For the end user it might be nicer, though development wise any changes made in the NetworkIsValidClientName require this message to be changed as it might otherwise be out-of-date. Having said that, entering a (currently) invalid name already requires the server owner knowing that you need to use two double quotes to pass an empty name, and double quotes with spaces between to get a name starting with a space.
To make matters "worse", I'd argue most changes of other people's nicks are done via rcon which requires the command to be put in double quotes, so you cannot even use double quotes for the client name with rcon making it impossible to trigger it with the current implementation of de validation.
So the only case where this error could be triggered is when you are hosting a game yourself (non-dedicated) and want to change the nick of some other client, which to me sounds like the demographic that does not know about the console.